### PR TITLE
[WIP] Fix a case where the upgrade job was stuck if its image was changed a bit late in the spec override section

### PR DESCRIPTION
**Pull request by aliok**
Fixes #

## Proposed Changes

* Fix a case where the upgrade job was stuck if its image was changed a bit late in the spec override section
*
*

**Release Note**

```release-note

```

### CODE REVIEW

**Comment by jcrossley3** [comment]
@aliok As discussed in slack, I think this is accounting for a situation that will only occur due to our downstream architecture, so let's handle it there.

File: pkg/reconciler/knativeeventing/knativeeventing.go [hunk 1]
```diff
@@ -123,6 +125,7 @@ func (r *Reconciler) reconcile(ctx context.Context, ke *eventingv1alpha1.Knative
 		r.ensureFinalizer,
 		r.initStatus,
 		r.install,
+		r.checkJobs,
```
Comment (houshengbo) [inline pkg/reconciler/knativeeventing/knativeeventing.go:128]: I actually would like to see a new transformer to convert the Job named `storage-version-migration`.
@jcrossley3 what do u think?

**Comment by houshengbo** [comment]
https://github.com/knative/operator/pull/180
I opened a PR to rename job name. I do not think I am doing the same work as Ali. Correct?

**Comment by aliok** [comment]
@houshengbo 
Thanks for the heads up.
It is a different problem I think.
I was testing something in this PR to fix a problem downstream at OpenShift side.

For the curious, it was about where our downstream stuff was too late to change the image overrides in the spec and thus the upstream operator was complaining that it cannot change the image of the job.
We decided to tackle this problem downstream though.

**Pull request closed without merging**